action.yaml

```diff
@@ -78,14 +84,22 @@ runs:
 
         echo Scanning Docker container ${CONTAINER_IMAGE} with Mondoo Client...
         mondoo scan --detect-cicd docker://${CONTAINER_IMAGE} --config mondoo.json --score-threshold ${{ inputs.score_threshold }} --output ${{ inputs.output_format }} ${{ inputs.extra_args }}
-      if: inputs.scan_type == 'docker_image_from_dockerfile' || inputs.scan_type == 'docker_image'
-    - name: Scan configs
+      if: inputs.scan_type == ^docker_image$|^docker_image_from_dockerfile$ && inputs.scan_type != ^k8s$|^terraform$
+    # Only run on k8s manifests, where ```--path``` is required for manifests.
+    - name: Scan k8s manifest
       shell: bash
       run: |
         echo Scanning ${{ inputs.path }} with Mondoo Client...
         mondoo scan --detect-cicd ${{ inputs.scan_type }} --path ${{ inputs.path }} --config mondoo.json --score-threshold ${{ inputs.score_threshold }} --output ${{ inputs.output_format }} ${{ inputs.extra_args }}
-      if: inputs.scan_type != 'docker_image_from_dockerfile' && inputs.scan_type != 'docker_image'
-          
+      if: inputs.scan_type == 'k8s' && inputs.scan_type != ^docker_image$|^docker_image_from_dockerfile$|^terraform$
+    # Only run on Terraform manifests, where a path is required and ```--path``` isn't supported.
+    - name: Scan Terraform configuration files
+      shell: bash
+      run: |
+        echo Scanning ${{ inputs.path }} with Mondoo Client...
+        mondoo scan --detect-cicd ${{ inputs.scan_type }} ${{ inputs.path }} --config mondoo.json --score-threshold ${{ inputs.score_threshold }} --output ${{ inputs.output_format }} ${{ inputs.extra_args }}
+      if: inputs.scan_type == 'terraform' && inputs.scan_type != ^docker_image$|^docker_image_from_dockerfile$|^k8s$
```

Comment:
could this be easier? `if: inputs.scan_type == 'terraform' `

Reply:
Yeah it could, I was simply modifying what was there. The original code had both listed the scan_types it supported and the scan_types it didn't support.

If we're comfortable enough only listing the scan_types it should support, I'm fine with removing the second part!